### PR TITLE
provide name override field for flowlog definition

### PR DESCRIPTION
**Pull request by drewmullen**
closes: #126 

Little bit of a shotgun approach.. could just change the iam role but the resources are logically grouped so i figured this was more clear

```shell
$ go test -run  TestExamplesPublicPrivateCWLogs -timeout 45m
...

TestExamplesPublicPrivateCWLogs 2023-08-11T13:40:43-04:00 logger.go:66: 
PASS
ok      github.com/aws-ia/terraform-aws-vpc     235.733s
```

### CODE REVIEW

**Comment by drewmullen** [comment]
If merged should be 4.3.0

File: test/examples_public_private_test.go [hunk 1]
```diff
@@ -11,16 +11,22 @@ func TestExamplesPublicPrivateCWLogs(t *testing.T) {
...
+		},
 	}
 
 	defer terraform.Destroy(t, terraformOptions)
 	terraform.InitAndApply(t, terraformOptions)
 	terraform.ApplyAndIdempotent(t, terraformOptions)
 
-	publicTagsLength := terraform.Output(t, terraformOptions, "public_subnets_tags_length")
-	assert.Equal(t, "2", publicTagsLength)
-	privateTagsLength := terraform.Output(t, terraformOptions, "private_subnets_tags_length")
```
Comment (andrew-glenn) [inline test/examples_public_private_test.go:22]: Was this removed intentionally?

Comment (drewmullen) [reply test/examples_public_private_test.go:22]: good catch. meant to add those back after running the test i intended. adding now, 1s

**Comment by drewmullen** [comment]
This topic is being discussed at the provider level as well. some resources have VERY strict restrictions on name_prefixes: https://github.com/hashicorp/terraform-provider-aws/issues/32838

**Comment by drewmullen** [comment]
@andrew-glenn any chance this can get merged and released?

**Review by pablo19sc — APPROVED** [review]
Checked flow logs example - all good

Screenshot 2023-10-29 at 15 15 48

**Review by tbulding — APPROVED** [review]

**Pull request merged**